akka-actor/src/main/scala-2.12/akka/util/ByteString.scala

```diff
@@ -1045,37 +1047,45 @@ final class ByteStringBuilder extends Builder[Byte, ByteString] {
     _length += 1
     this
   }
-
+  def ++=(bytes: ByteString): this.type = {
+    if (bytes.nonEmpty)
+      bytes match {
+        case b: ByteString1C =>
+          clearTemp()
+          _builder += b.toByteString1
+          _length += b.length
+        case b: ByteString1 =>
+          clearTemp()
+          _builder += b
+          _length += b.length
+        case bs: ByteStrings =>
+          clearTemp()
+          _builder ++= bs.bytestrings
+          _length += bs.length
+      }
```

Comment:
`clearTemp()` happens for all cases so perhaps pull out and do before patmatch?

Reply:
:+1:, why not. Done